server/mailet/mailets/src/test/java/org/apache/james/transport/matchers/dlp/DlpTest.java

```diff
@@ -102,6 +103,18 @@ void matchSenderShouldReturnRecipientsWhenEnvelopSenderMatches() throws Exceptio
         assertThat(dlp.match(mail)).contains(RECIPIENT1);
     }
 
+    @Test
+    void nullSenderShouldBeIgnored() throws Exception {
```

Comment:
I'm not sure about this fix. What if my DLP rule goal is actually catching null sender ? Or any sender ?
I think that null sender should be matched when we want to match "any sender", don't you think ?

Reply:
It's a bit more *foudamental* than what you think...

The problem is that no domain is attached to null-sender: you don't know which rules to load.

Reply:
What to do is more a business decision, let's ask our Product Owner

Reply:
let's try to be simple. DLP are not to protect from an attack, but to prevent users to do bad things and not to leak any data.
So we cannot assume an user will forge a null sender just to be able to send confidential data. If he really wants it, he will just put this in an attachment, encode it or whatever.
So I think we can safely ignore null senders as we are not able to determinate which rules applies for this user.

Reply:
how can I be a nullsender if I need to be authenticated (through JWT tokens or login/pass) to send emails ?

Reply:
Well, you know : SMTP, ops, server configuration, forged emails, etc.
The question is more "does DLP intended to protect from an attack or from a user mistake ?"

Reply:
For the record, I've found this bug looking at our production's logs.

Reply:
Even if you decide that nullSender should be checked by DLP `by definition`  no domain is attached to this sender.

No sender domain means that it is impossible to choose a set of rules to load.

I do agree with Michael that authenticated users should not be able to use NullSender, that would IMO solve the issue.